api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1694,19 +1694,73 @@ type AWSServiceEndpoint struct {
+	//
+	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud;AzureStackCloud
 	// +kubebuilder:default="AzurePublicCloud"
-	Cloud             string `json:"cloud,omitempty"`
-	Location          string `json:"location"`
+	Cloud string `json:"cloud,omitempty"`
+
+	// Location is Azure region in which the Azure cloud infrastructure resources will be created
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location is immutable"
+	// +immutable
+	Location string `json:"location"`
+
+	// ResourceGroupName is the name of the resource group where any cloud infrastructure resources
+	// created by the Hosted Cluster are to be placed
+	//
+	// +kubebuilder:default:=default
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
+	// +immutable
 	ResourceGroupName string `json:"resourceGroup"`
-	VnetName          string `json:"vnetName"`
-	VnetID            string `json:"vnetID"`
-	SubnetName        string `json:"subnetName"`
-	SubscriptionID    string `json:"subscriptionID"`
-	MachineIdentityID string `json:"machineIdentityID"`
-	SecurityGroupName string `json:"securityGroupName"`
+
+	// VnetName is the name of the VNET to place created subnets and VMs into
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetName is immutable"
+	// +immutable
+	VnetName string `json:"vnetName"`
+
+	// VnetID is the ID of the VNET to place created subnets and VMs into
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetID is immutable"
+	// +optional
+	// +immutable
+	VnetID string `json:"vnetID,omitempty"`
+
+	// SubnetName is the name of the default subnet in the Hosted Cluster; this is also the subnet where
+	// the Azure CCM will place the load balancer to be used for node egress
+	//
+	// +kubebuilder:default:=default
+	// +kubebuilder:validation:Required
+	SubnetName string `json:"subnetName"`
+
+	// SubscriptionID is a unique identifier for an Azure subscription used to manage resources
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SubscriptionID is immutable"
+	// +immutable
+	SubscriptionID string `json:"subscriptionID"`
+
+	// MachineIdentityID is used as the user-assigned identity to be assigned to the VMs
+	//
+	// +optional
+	MachineIdentityID string `json:"machineIdentityID,omitempty"`
+
+	// SecurityGroupName is the name of the security group to be used for the Hosted Cluster
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupName is immutable"
+	// +optional
+	// +immutable
+	SecurityGroupName string `json:"securityGroupName,omitempty"`
```

Comment:
Thanks. I think all of this set of restrictions would be useful to have them explicitly documented for the consumers of the API. Specially if the attributes are full resource IDs instead of names